R/bitbucket.R

```diff
@@ -3,11 +3,17 @@ isBitbucketURL <- function(url) {
 }
 
 canUseBitbucketDownloader <- function() {
-  all(packageVersionInstalled(devtools = "1.9.1", httr = "1.0.0"))
+  (all(packageVersionInstalled(httr = "1.0.0")) &
+     !is.null(bitbucket_user(quiet = TRUE)) &
+     !is.null(bitbucket_pwd(quiet = TRUE)))
 }
 
 bitbucketDownload <- function(url, destfile, ...) {
-  onError(1, bitbucketDownloadImpl(url, destfile, ...))
+  tryCatch(
+    bitbucketDownloadImpl(url, destfile, ...),
+    error = function(e) {
+      stop(sprintf("Bitbucket request failed with %s", e), call. = FALSE)
```

Comment:
```suggestion
      stop(sprintf("Bitbucket request failed with %s", conditionMessage(e)), call. = FALSE)
```

Reply:
I guess this depends on whether you want the error prefix here; e.g.

```
> err <- tryCatch(stop("oh no", call. = FALSE), error = identity)
> sprintf("%s", err)
[1] "Error: oh no\n"
> sprintf("%s", conditionMessage(err))
[1] "oh no"
```

so the existing formulation is probably fine.

Reply:
Because these eventually get wrapped again, it's probably better to use `conditionMessage`.

Reply:
After some more testing, I've decided to leave it as-is because we get more context for things like the original `devtools::github_pat` error:

```
Reason: Error: GitHub request failed: Error in get(name, envir = asNamespace(pkg), inherits = FALSE): object 'github_pat' not found
```

Reply:
I've slightly tweaked the message, but am not using `conditionMessage`.